Remove THCTensorMathReduce.cuh

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66391
* #66480
* #66390
* __->__ #66389
* #66388

Differential Revision: [D31547711](https://our.internmc.facebook.com/intern/diff/D31547711)

peterbell10 committed:
Remove THCTensorMathReduce.cuh

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31547711).

peterbell10 committed:
Update on "Remove THCTensorMathReduce.cuh"


Differential Revision: [D31547711](https://our.internmc.facebook.com/intern/diff/D31547711)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31547711).
